### PR TITLE
Add find functionality

### PR DESCRIPTION
**Pull request by HistoireDeBabar**
Hey, not sure if this will be useful for anyone but I found myself needing a find function to return all matching emoji's.

### CODE REVIEW

File: test/emoji.js [hunk 1]
```diff
@@ -77,4 +77,40 @@ describe("emoji.js", function () {
     should.exist(coffee);
     coffee.should.be.exactly('☕️');
   });
+
+  describe("find(str)", function () {
```
Comment (omnidan) [inline test/emoji.js]: the function is called `search` above, but still called `find` in the tests

File: lib/emoji.js [hunk 2]
```diff
@@ -9,6 +9,26 @@ require('string.prototype.codepointat');
...
+ * @return {string}
+ */
+var trim = function(str) {
+  var colonIndex = str.indexOf(':');
+  if (colonIndex > -1) {
+    // :emoji: (http://www.emoji-cheat-sheet.com/)
+    if (colonIndex === str.length - 1) {
+      str = str.substring(0, colonIndex);
+      return trim(str);
+    } else  {
```
Comment (omnidan) [inline lib/emoji.js]: remove the second space after `else`

File: test/emoji.js [hunk 3]
```diff
@@ -77,4 +77,40 @@ describe("emoji.js", function () {
...
+    it("should return partially matched emojis", function () {
+      var matchingEmojis = emoji.search("cof");
+      var coffee = matchingEmojis[0];
```
Comment (omnidan) [inline test/emoji.js]: I wouldn't rely on coffee being the first result here (in case more emoji get added later), check if one of the matches was coffee instead

**Comment by omnidan** [comment]
@HistoireDeBabar can you also add this feature to the README.md?

**Comment by HistoireDeBabar** [comment]
Hey, NP! Thanks for the feedback, I've updated the PR!

**Comment by omnidan** [comment]
@HistoireDeBabar Thanks a lot for the quick response! :grin: I noticed another small thing with the tests - I think most would still pass even if the search function always returns an empty array. Maybe you should check if there is at least one result first before doing the `forEach` checks.

**Comment by HistoireDeBabar** [comment]
Ah you're completely right! I've added an expectation to ensure the array has a greater length than zero, which should suffice. Cheers @omnidan

**Comment by omnidan** [comment]
@HistoireDeBabar perfect! thanks a lot :grin:

**Comment by omnidan** [comment]
released `1.5.0` with this change
